src/lib/update-alternatives/UI/main_dialog.rb

```diff
@@ -74,6 +91,27 @@ def map_alternatives_items
       end
     end
 
+    def filtered_alternatives
+      alternatives = @alternatives_list.select { |a| Regexp.new(@search).match(a.name) }
```

Comment:
and btw running this search + multi_choice break correct indexes. Problem is that select is done before each_with_index, so wrong index is used.

I expect code can loop like:
```
alternatives = @alternatives_list.each_with_index
alternatives.select { |v, _i| v.choices.length > 1 } if @multi_choice_only
alternatives.select { |a, _i| Regexp.new(@search).match(a.name) } unless @search.empty?

alternatives
```